src/api/api.js

```diff
@@ -0,0 +1,148 @@
+const express = require('express');
+const bodyParser = require('body-parser');
+
+const logger = require('../logger');
+
+const ApiHandler = require('./api-handler');
+
+/**
+ * Ghost Pinbot API implementation
+ */
+class Api {
+  constructor(apiHandler) {
+    this.port = process.env.PORT || 8081;
+    this.handler = apiHandler || new ApiHandler();
+  }
+
+  /**
+   * Starts the server
+   * @return {Promise<void>}
+   */
+  async start() {
+    this.app = express();
+    this.app.use(bodyParser.urlencoded({ extended: false }));
+    this.app.use(bodyParser.json());
+
+    // rooms
+    this.app.get('/api/v0/rooms', this.listRooms.bind(this));
+    this.app.post('/api/v0/rooms', this.attach.bind(this));
+    this.app.delete('/api/v0/rooms', this.detach.bind(this));
+
+    // peers
+    this.app.get('/api/v0/peer', this.info.bind(this));
+
+    // callbacks
+    this.app.post('/api/v0/callback', this.callback.bind(this));
+
+    this.app.listen(this.port, () => logger.info(`Server started on port ${this.port}`));
+  }
+
+  /**
+   * Peer info route
+   */
+  info(req, res) {
+    logger.debug('get peer information');
+    return res.status(200)
+      .send(this.handler.info);
+  }
+
+  /**
+   * List attached rooms
+   */
+  listRooms(req, res) {
+    logger.debug('list rooms');
+    const rooms = this.handler.listRooms();
+    return res.status(200)
+      .send(rooms);
+  }
+
+  /**
+   * Attach to space/thread
+   */
+  async attach(req, res) {
+    const { space, thread } = req.body;
+
+    try {
+      logger.debug(`attach to space ${space} and thread ${thread}`);
+
+      const result = await this.handler.joinThread(space, thread);
+      return res.status(200)
+        .send(result);
+    } catch (e) {
+      return res.status(200)
```

Comment:
Is is returning a 200 on error for a specific reason please? It might break some client logic expecting errors to have 4xx codes.

Reply:
nope, just an error :) nice catch!